feat: add build config configurator

cdrage opened the pull request:
feat: add build config configurator

### What does this PR do?

* Adds automated way to create a build config json file
* Added to build page

### Screenshot / video of UI

<!-- If this PR is changing UI, please include
screenshots or screencasts showing the difference -->
![Screenshot 2024-11-18 at 3 44 47 PM](https://github.com/user-attachments/assets/a48fdd62-f3aa-4a31-b088-a637632c5fea)


https://github.com/user-attachments/assets/bbf18809-620f-46f5-ad5e-a02c8df69fd0



### What issues does this PR fix or reference?

<!-- Include any related issues from Podman Desktop
repository (or from another issue tracker). -->

Closes https://github.com/podman-desktop/extension-bootc/issues/827

### How to test this PR?

<!-- Please explain steps to reproduce -->

Use build config and configure / create your own custom build config
   with users, filesystem, etc.

cdrage commented:
Discussed with @mrguitar on Slack, follow up issues after this PR:

https://github.com/podman-desktop/extension-bootc/issues/1027 and https://github.com/podman-desktop/extension-bootc/issues/1028

deboer-tim commented on packages/frontend/src/Build.svelte:
```diff
@@ -703,15 +882,22 @@ $: if (availableArchitectures) {
-                      externalRef="https://github.com/osbuild/bootc-image-builder?tab=readme-ov-file#-build-config"
-                      >bootc-image-builder documentation</Link
-                    >.
+                    This will override any above user-specific input and use the supplied file only.
```
The text at the top only talks about filling in the fields, then at the bottom of the form you find out you can also provide a file. Feels like maybe there should be two radio buttons like 'Create new build config' and 'Supply existing build config file' so that it's clear you have either option from the start?

deboer-tim commented on packages/frontend/src/Build.svelte:
```diff
@@ -684,14 +762,115 @@ $: if (availableArchitectures) {
-              {#if showAdvanced}
+              {#if showBuildConfig}
+                <p class="text-sm text-[var(--pd-content-text)] mb-2">
+                  Supplying the following fields will create a build config file that contains the build options for the
```
It never says _where_ the build config will be generated. I think this would help (e.g. say we put it in the folder above), so that I could use this once to create it, then maybe edit it outside of Podman Desktop, then supply the same file next time I build.

deboer-tim reviewed (commented):
Very nice improvement 👍🏼, worked well in testing but just had some usability suggestions.

cdrage commented:
@deboer-tim @axel7083 

I've updated the PR with the following:
- Config.json is not created in the same directory as the output folder, so you can further edit it, etc.
- Updated the UI a bit so file and interactive mode are in different sections (we can improve it post-this-PR @deboer-tim for radio mode after I sync up with @MariaLeonova / UI/UX team)

![Screenshot 2024-12-16 at 11 23 12 AM](https://github.com/user-attachments/assets/33d861b4-cf1c-462a-9c7f-fb73417fdc23)
